client/src/pages/DirectoryPage/DirectoryPage.tsx

```diff
@@ -1,9 +1,23 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { useAppSelector } from "../../app/hooks";
+import axios from "axios";
```

Comment:
wondering if there would be value in creating custom axios instances where we could set reusable configs like baseUrl, withCredentials, headers etc. Could setup interceptors for responses too to simplify error handling